Yoast/wordpress-seo
# Add error boundary around PrimaryTaxonomyPicker

## Summary

This PR can be summarized in the following changelog entry:

* Fall back to original taxonomy picker if an error occurs in the primary taxonomy picker.

## Relevant technical choices:

* This creates a wrapper around the PrimaryTaxonomyPicker and fallback to the original Taxonomy picker if an error occurs in the Primary Taxonomy picker, as in #10850.
Obviously this is not a real solution to the problem, but it will lead to a better user experience.

## Test instructions

This PR can be tested by following these steps:

* Enable Gutenberg and the Classic Editor plugin.
* Under `Settings` > `Writing` enable Gutenberg when the classic editor plugin is enabled.

* Create a new post or page in Gutenberg.
* Open `Settings` (Gearwheel) > `Document`.
* Make sure that the taxonomy picker can be used, but the primary term picker is not present.
**Note**: The error ( about `getEditedPostAttribute`) is still thrown, but is handled internally by the component.
* To make sure it actually tried to render the primary taxonomy picker, add a `console.log`statement in `componentDidCatch` in `PrimaryTaxonomyPicker.js`. [Andrea edit] I guess it's `PrimaryTaxonomyFilter.js`.

* Now disable the classic editor plugin and make sure the primary taxonomy picker works as intended in "regular" Gutenberg.

## Quality assurance


Fixes #10850
